src/apps/recommender/RecommendMaterial.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from "react";
+import { FC } from "react";
+import { useDispatch } from "react-redux";
+import ButtonFavourite, {
+  ButtonFavouriteId
+} from "../../components/button-favourite/button-favourite";
+import { Cover } from "../../components/cover/cover";
+import { Work } from "../../core/utils/types/entities";
+import { getContributors } from "../../core/fetchers/helpers";
+import { getManifestationPid } from "../../core/utils/helpers/general";
+import { TypedDispatch } from "../../core/store";
+import { guardedRequest } from "../../core/guardedRequests.slice";
+import { constructMaterialUrl } from "../../core/utils/helpers/url";
+import { Link } from "../../components/atoms/link";
+import { useUrls } from "../../core/utils/url";
+
+export interface RecommendMaterialProps {
+  work: Work;
+}
+
+const RecommendMaterial: FC<RecommendMaterialProps> = ({
+  work: {
+    titles: { full: title },
+    creators,
+    workId,
+    manifestations: { all: manifestations }
+  }
+}) => {
+  const dispatch = useDispatch<TypedDispatch>();
+  const { materialUrl } = useUrls();
+  const materialFullUrl = constructMaterialUrl(materialUrl, workId);
+
+  // Create authors string
+  let authors = null;
+  const inputContributorsArray = creators?.map(({ display }) => display);
+  if (inputContributorsArray) {
+    authors = getContributors(inputContributorsArray);
+  }
+
+  // For retrieving cover
+  const manifestationPid = getManifestationPid(manifestations);
+
+  const addToListRequest = (materialId: ButtonFavouriteId) => {
+    dispatch(
+      guardedRequest({
+        type: "addFavorite",
+        args: { materialId },
+        app: "search-result"
```

Comment:
I think you need the specify the recommender app here, to make the system work in that context only.
So it should be: `app: "recommender"` and specified [here](https://github.com/reload/dpl-react/blob/730d8ac11dcb95a86bb625cd8f36f93e7a873add/src/core/utils/types/ids.ts#L6) as well.
And you need to wrap the app in the guardedApp HOC like the [search-result app is](https://github.com/reload/dpl-react/blob/730d8ac11dcb95a86bb625cd8f36f93e7a873add/src/apps/search-result/search-result.entry.tsx#L76-L78).

Reply:
done